xarray/tests/test_quantities.py

```diff
@@ -0,0 +1,121 @@
+import numpy as np
+import pytest
+
+from xarray import DataArray
+
+try:
+    import quantities as pq
+    has_quantities = True
+except ImportError:
+    has_quantities = False
+
+pytestmark = pytest.mark.skipif(
+    not has_quantities,
+    reason="requires python-quantities",
+)
+
+
+def equal_with_units(a, b):
+    a = a if not isinstance(a, DataArray) else a.data
+    b = b if not isinstance(b, DataArray) else b.data
+
+    return (
+        (hasattr(a, "units") and hasattr(b, "units"))
+        and a.units == b.units
+        and np.allclose(a.magnitude, b.magnitude)
+    )
+
+
+@pytest.fixture
+def data():
```

Comment:
This is a minor point, but I would recommend against using pytest fixtures for test data. (Though we do use this style currently in xarray in a few places.)

Fixtures are great for setup/teardown (e.g., cleaning up temporary files), but when you test depends on data values, it's best if the data values can be found in the test itself. Otherwise, the logic in a test is not self-contained, which means you have to understand a much bigger context of code.

All this is true for even helper functions, but pytest's fixtures are even more magical:
- They use magic name matching instead of Python's lookup rules (https://github.com/pytest-dev/pytest/issues/3834)
- They can add arbitrary teardown logic, beyond merely creating an object.

So in summary:
- Prefer creating test data in a test itself, if your test relies on any aspect of the data values.
- If you do want to re-use constructors, prefer using normal helper functions to fixtures. These are easier to understand.

Reply:
right, so I used helper functions instead